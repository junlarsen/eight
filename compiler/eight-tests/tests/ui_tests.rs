//! Snapshot tests for the user interface reported by the compiler.
//!
//! These tests are performed by running the compiler for each of the files in the `ui` directory,
//! recursively.
//!
//! The goal is for each of the tests to fail, and for us to be able to view the output of the
//! compiler, ensuring that we're getting useful and precise error messages.

use insta_cmd::get_cargo_bin;
use std::process::Command;

#[test]
fn test_compiler_ui_tests() {
    insta::glob!("ui/**/*.test", |path| {
        let mut cmd = Command::new(get_cargo_bin("eightc"));
        let relative = path.strip_prefix(env!("CARGO_MANIFEST_DIR")).unwrap();
        insta_cmd::assert_cmd_snapshot!(cmd.arg(relative));
    });
}
