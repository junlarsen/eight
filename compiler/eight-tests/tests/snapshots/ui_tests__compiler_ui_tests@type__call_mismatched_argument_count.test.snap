---
source: compiler/eight-tests/tests/ui_tests.rs
info:
  program: eightc
  args:
    - tests/ui/type/call_mismatched_argument_count.test
input_file: compiler/eight-tests/tests/ui/type/call_mismatched_argument_count.test
snapshot_kind: text
---
success: false
exit_code: 1
----- stdout -----

----- stderr -----
Error: sema::function_type_mismatch

  × function types do not take the same number of arguments
   ╭─[tests/ui/type/call_mismatched_argument_count.test:6:11]
 5 │ fn test() {
 6 │   let k = apply_to_me(123);
   ·           ─────┬─────
   ·                ╰── the function has type fn(i32, bool) -> i32
 7 │ }
   ╰────
